flax/training/checkpoints.py

```diff
@@ -121,7 +121,8 @@ def restore_checkpoint(ckpt_dir, target, step=None, prefix='checkpoint_'):
 
   Args:
     ckpt_dir: str: directory of checkpoints to restore from.
-    target: matching object to rebuild via deserialized state-dict.
+    target: matching object to rebuild via deserialized state-dict. If None,
+      the deserialized state-dict is returned as-is.
```

Comment:
Maybe add a comment that usually you would want to pass a `target` to make sure the deserialized object has the expected form. At least I think that was the original intention of `target`.